src/Context/CommonContext.php

```diff
@@ -391,18 +391,19 @@ public function iShouldNotSeeElementByCss($selector): void
      */
     public function thenIShouldSeePage(string $url): void
     {
+        $current_url = $this->getSession()->getCurrentUrl();
         $windowNames = $this->getSession()->getWindowNames();
         if (count($windowNames) > 1) {
             $this->switchToNewTab($windowNames);
+            $current_url = $this->getSession()->getCurrentUrl();
```

Comment:
Why are you overriding $current_url in this step.

Reply:
if it more than 1 window it take the new window url, otherwise it take the main window url.

Reply:
What is the use case for this method?
We already have a method to verify the page url. If you are looking to verify the url of the new window then you should not have if condition.